dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpTransport.java

```diff
@@ -56,6 +58,15 @@ public McpTransport(HttpServletRequest req, HttpServletResponse res, Jsonb jsonb
         writer = res.getWriter();
     }
 
+    /**
+     * Returns the current session associated with this transport, or null if no valid session was provided.
+     *
+     * @return the {@link SessionInfo} for this transport, or null if not available
+     */
+    public SessionInfo getSession() {
+        return sessionInfo;
+    }
+
```

Comment:
Looks good, but please move this down the file next to the other setter methods.